pkg/messaging/sns.go

```diff
@@ -23,27 +23,31 @@ func NewSNS(config *Config, params ...string) (*sns.SNS, string, error) {
 
 	sess, err := session.NewSession(awscfg)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
 	var client *sns.SNS
 	client = sns.New(sess)
 	if client == nil {
-		return nil, "", err
+		return nil, err
 	}
+	return client, nil
+}
 
+// TopicArn returns topicArn for a specific topic display name
+func TopicArn(client *sns.SNS, topic string) (string, error) {
 	topicArn := ""
-	if len(params) > 0 {
+	if len(topic) > 0 {
```

Comment:
you don't need this if clause anymore...

actually, you should have a check above this that is 

```
if len(topic) < 1 {
return "", errors.New("Invalid empty parameter topic")
}
```

then you can remove this if wrapping the lookup

also, this is a great case for named returns ... so if your return was `(topicARN string, err error)`

then you don't need to initialize an empty topicARN, and you can just set err = whatever, and return topicARN, err